### PR TITLE
My Q1.1 and Q1.3 solutions

### PR DESCRIPTION
**Pull request by ele7ija**

### CODE REVIEW

File: ch01/01_is_unique.go [hunk 1]
```diff
@@ -30,3 +32,21 @@ func IsUniqueVanilla(s string) bool {
...
+
+// IsUniqueBits determines if a string has all unique chars (only works for ASCII)
+// This solution doesn't utilize any additional data structures and has O(n) time complexity
+func IsUniqueBits(s string) bool {
+
+	s = strings.ToLower(s)
+	var vector int32
+	sBytes := []byte(s) // only for ASCII
+	for _, sByte := range sBytes {
+		index := sByte - 'a'
```
Comment (krlv) [inline ch01/01_is_unique.go]:
```suggestion
	for _, rune := range s {
		index := rune - 'a'
```
In Go [a string is a read-only slice of bytes](https://go.dev/blog/strings#what-is-a-string-h2); we don't need to explicitly convert it to byte slice to pass it to `range`.

File: ch01/03_urlify.go [hunk 2]
```diff
@@ -55,3 +57,10 @@ func URLify(s string) string {
 
 	return string(runes)
 }
+
+func UrlifyRegex(s string) string {
```
Comment (krlv) [inline ch01/03_urlify.go]:
```suggestion

// UrlifyRegex returns the s string with HTML-encoded space characters.
func UrlifyRegex(s string) string {
```
It's an exported method; let's add some comments to it.

File: ch01/01_is_unique.go [hunk 3]
```diff
@@ -30,3 +32,21 @@ func IsUniqueVanilla(s string) bool {
 
 	return true
 }
+
+// IsUniqueBits determines if a string has all unique chars (only works for ASCII)
+// This solution doesn't utilize any additional data structures and has O(n) time complexity
+func IsUniqueBits(s string) bool {
+
+	s = strings.ToLower(s)
```
Comment (krlv) [inline ch01/01_is_unique.go:40]: I think we can drop it since the bit mask approach will only work for case insensitive ascii strings.

**Review by krlv — COMMENTED** [review]
@ele7ija thanks for you contribution! 
PR looks great, left some minor suggestions.

File: ch01/01_is_unique.go (hunk 3 above)
Comment (ele7ija) [reply ch01/01_is_unique.go:40]: I actually got this idea from a hint related to this question:

> Hint #117: Could a bit vector be useful?

**Comment by krlv** [comment]
Thanks @ele7ija !

**Pull request merged**